opennms-config/src/main/java/org/opennms/netmgt/config/DiscoveryConfigFactory.java

```diff
@@ -525,18 +525,29 @@ private void clearUrlSpecifics() {
     /**
      * <p>isExcluded</p>
      *
-     * @param address a {@link java.net.InetAddress} object.
+     * @param address a {@link InetAddress} object.
+     * @param location
      * @return a boolean.
      */
     @Override
-    public boolean isExcluded(final InetAddress address) {
+    public boolean isExcluded(final InetAddress address, String location) {
         getReadLock().lock();
         try {
-            final List<ExcludeRange> excludeRange = getConfiguration().getExcludeRanges();
-            if (excludeRange != null) {
+            final List<ExcludeRange> excludeRanges = getConfiguration().getExcludeRanges();
+            // Get exclude ranges from definitions. Assign location from definition.
+            getConfiguration().getDefinitions().forEach(def -> {
```

Comment:
Shouldn't this be performed when the configuration is loaded/reloaded as opposed to being done on every call to `isExcluded`?